roxie/roxiemem/roxiemem.cpp

```diff
@@ -1955,19 +1983,22 @@ class HugeHeaplet : public Heaplet
 
     virtual void noteReleased(const void *ptr)
     {
+        //NORE: See comment on FixedSizeHeaplet::noteReleased() regarding the memory order operands
+
         //If rowCount == 1 then this must be the last reference - avoid a locked operation.
         //rowCount is not used once the heaplet is known to be freed, so no need to ensure rowCount=0
-        if (atomic_read(&rowCount) == 1 || atomic_dec_and_test(&rowCount))
+        if (rowCount.load(std::memory_order_relaxed) == 1 || rowCount.fetch_sub(1, std::memory_order_release) == 1)
```

Comment:
This feels like it's changed the semantics - dec and test would return 0 in the case that fetch_sub returns 1 ?